prereise/gather/griddata/hifld/data_access/load.py

```diff
@@ -0,0 +1,94 @@
+import json
+import os
+import tempfile
+from io import BytesIO
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import pandas as pd
+
+from prereise.gather.griddata.hifld.const import abv2state  # noqa F401
+
+
+def get_hifld_power_plants(path):
+    """Read the HIFLD Power Plants CSV file and keep operational plants located in
+    contiguous states.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- operational power plant in contiguous states.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'OP' and STATE in @abv2state")
+
+
+def get_hifld_generating_units(path):
+    """Read the HIFLD Generating Units CSV file and keep operational plants located in
+    contiguous states.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- operational generating units in contiguous states.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'OP' and STATE in @abv2state")
+
+
+def get_hifld_electric_substations(path):
+    """Read the HIFLD Electric Substations CSV file and keep in service substations
+    located in contiguous states and connected to at least one transmission line.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- in service electric substations in contiguous
+        states that are connected to at least one electric power transmission line.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'IN SERVICE' and STATE in @abv2state and LINES != 0")
+
+
+def get_hifld_electric_power_transmission_lines(path):
```

Comment:
We are assuming there are other data sources in CSV format for transmission lines comparing with what we do in 
`get_hifld_electric_power_transmission_lines_topology`, right? Then should we rename these two functions into something like `get_hifld_electric_power_transmission_lines_from_csv` and `get_hifld_electric_power_transmission_lines_from_zip`?

Reply:
In fact, the hope is to have a combined data frame. The two data frames differ by one column, `coordinates` but for some reason we need to understand the content of some of the columns is different:
```
>>> (topology.drop("coordinates", axis=1) == lines).all()
OBJECTID         True
ID              False
TYPE             True
STATUS           True
NAICS_CODE      False
NAICS_DESC       True
SOURCE          False
SOURCEDATE       True
VAL_METHOD       True
VAL_DATE         True
OWNER            True
VOLTAGE         False
VOLT_CLASS       True
INFERRED         True
SUB_1            True
SUB_2            True
SHAPE_Length    False
dtype: bool
```
I would not do any refactoring now since we will hopefully end up with a single function reading both files and returning a single data frame.

Reply:
Make sense.